spike_main/spike.cc

```diff
@@ -445,9 +452,28 @@ int main(int argc, char** argv)
   }
 #endif
 
+  if (cfg.explicit_hartids) {
```

Comment:
It seems that nobody outside of this code in `spike.cc` cares about `explicit_hartids`? Can this be a local variable here in `main()` instead of in `cfg_t`?

Reply:
This is analogous to the `overridden()` method on `cfg_arg_t<T>`. Once we are merging with a parsed DTB file, I think we'll want to know whether the IDs have been explicitly or whether they are just inferred from the number of processors.

Reply:
Yeah, I guess the problem is that there are two different "knobs" that can be driven on the command line argument which both affect the same thing. I suspect there might be a few more similar special cases popping up before this has all settled down.